backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/DisplayController.java

```diff
@@ -0,0 +1,134 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/v1/displays")
+public class DisplayController {
+    private final DisplayService displayService;
+    private final AuthService authService;
+
+    public DisplayController(DisplayService displayService, AuthService authService) {
+        this.displayService = displayService;
+        this.authService = authService;
+    }
+
+    @PostMapping("/{tenantCode}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity createDisplay(@PathVariable final String tenantCode,
+                                        @RequestBody final DisplayCreateRequest request) {
+        if (!this.authService.canChangeTenant(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        try {
+            final Display createdDisplay = this.displayService.createDisplay(request.name(), request.branchId());
+            return ResponseEntity.ok().body(DisplayDto.fromEntity(createdDisplay));
+        } catch (final Exception exception) {
+            System.out.println(exception.getMessage());
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    @GetMapping("/{tenantCode}/{branchId}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity getDisplays(@PathVariable final String tenantCode,
+                                      @PathVariable final String branchId) {
+        if (!this.authService.canChangeTenant(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        try {
+            final Set<DisplayDto> displays = this.displayService.getDisplays(Long.parseLong(branchId)).stream()
+                    .map(DisplayDto::fromEntity)
+                    .collect(Collectors.toSet());
+            return ResponseEntity.ok().body(displays);
+        } catch (final Exception exception) {
+            System.out.println(exception.getMessage());
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    @GetMapping("/{tenantCode}")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_ADMIN', 'ROLE_BRANCH_ADMIN')")
+    public ResponseEntity getDisplaysByTenant(@PathVariable final String tenantCode) {
+        if (!this.authService.canChangeTenant(tenantCode)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        try {
+            final Set<DisplayDto> displays = this.displayService.getDisplaysByTenant(tenantCode).stream()
+                    .map(DisplayDto::fromEntity)
+                    .collect(Collectors.toSet());
+            return ResponseEntity.ok().body(displays);
+        } catch (final Exception exception) {
+            System.out.println(exception.getMessage());
```

Comment:
Do not use`System.out.println`. If you want to log something use a proper logger.